lib/fastlane/plugin/wpmreleasetoolkit/actions/common/check_for_toolkit_updates_action.rb

```diff
@@ -0,0 +1,77 @@
+require 'fastlane/action'
+require 'rubygems/command_manager'
+
+module Fastlane
+  module Actions
+    class CheckForToolkitUpdatesAction < Action
+      TOOLKIT_SPEC_NAME = 'fastlane-plugin-wpmreleasetoolkit'.freeze
+
+      def self.run(params)
+        updater = Gem::CommandManager.instance[:update]
+        installed_gems = updater.highest_installed_gems.select { |spec| spec == TOOLKIT_SPEC_NAME }
+        local_version = Gem::Version.new(installed_gems[TOOLKIT_SPEC_NAME].version)
+        UI.message("Currently using release toolkit version #{local_version.to_s.yellow}.")
+        UI.message("Checking for updates now...")
+        updates_needed = updater.which_to_update(installed_gems, [TOOLKIT_SPEC_NAME])
+
+        if updates_needed.empty?
+          UI.success('Your release toolkit is up-to-date! ✅')
+          return nil
+        end
+
+        # Return type of which_to_update differs before/after RubyGems 3.1.0, so normalize to always use Gem::NameTuple
+        updates_needed = Gem::NameTuple.from_list(updates_needed)
+        latest_version = updates_needed.find { |gem_info| gem_info.name == TOOLKIT_SPEC_NAME }.version
+
+        UI.message(['There is a new version '.yellow, latest_version.to_s.red, ' of the release toolkit!'.yellow].join)
+        non_breaking_update = Gem::Requirement.new(local_version.approximate_recommendation).satisfied_by?(latest_version)
```

Comment:
`Gem::Version#approximate_recommendation` will return something like `"~> 1.0"` (i.e. `~> <major>.<minor>`) for a `local_version` with a value of `1.0.1`.

Which allows us to then build a `Gem::Requirement` out of it – which will likely match what we plan to use in our `Pluginfile` anyway, but at least this works even if used a `tag:` or `branch:` in the `Pluginfile` too – and check if that `~> x.y` requirement is satisfied by the latest version. If not, that means that it is a major version bump (aka breaking change)